test/client.test.js

```diff
@@ -44,7 +44,15 @@ describe('client.js tests', () => {
                     return '123456';
                 }
             },
-            AdobeIOEvents: class AdobeIOEventsMock { },
+            AdobeIOEvents: class AdobeIOEventsMock { 
+                async getEventsFromJournal(url) {
+                    if(url === 'JOURNAL_NOT_READY') {
+                        throw Error('504');
```

Comment:
is this what the api actually throws when the journal is not ready? I believe i was seeing 500 internal server error in practice?

Reply:
Even if it's `504`, we should throw something else. 
504 is 504 Gateway Timeout - we are not acting as gateway or proxy here. I'd suggest to throw a "nicer" error (aka an error-exception object).

Reply:
is it just for readability reasons that you prefer `500` instead of `504` ? For unit test I only care that an error is thrown.  changing it to `throw Error('some_error')`  is even better as it spells out that my code does not care for the error type. wdyt ?

Reply:
Better to choose a proper error type/message here, as I don't think we care about the original error. The whole point is to handle it and poll until ready. RIght?

Reply:
You should probably use nock and actually mock the HTTP call to io events instead of just mocking the events library (especially since we are switching to a different events library soon...that way we just rely on the io events api call here)

Also, doesn't getEventsFromJournal return a JSON of events? https://github.com/adobe/asset-compute-events-client/blob/master/lib/events.js#L383-L412 

mocks should try to resemble the actual output of what is being mocked

Reply:
I started off with nock, but realized the [existing tests](https://github.com/adobe/asset-compute-client/blob/master/test/client.test.js#L38) would limit the use of nock. In my opinion, I am not unit testing `getEventsFromJournal` (which most likely has its test in https://github.com/adobe/asset-compute-events-client) so this seemed like an alternative.

@jdelbick in case you have pointers to any other code which uses to nock calls mocked by `mockRequire` do let me know. open to explore it.

Reply:
no need for nock if you use mockRequire... Thinking about it again, I agree using `mockRequire` to mock to asset-compute-events-client module is better than mocking the HTTP call inside that library (that is done inside the asset-compute-events-client tests).

However, I think you should aim mimick the behavior of `getEventsFromJournal` function:
1. in the error case, throw this [error](https://github.com/adobe/asset-compute-events-client/blob/master/lib/events.js#L409)
2. in the successful case, return an array of dummy events (or in the 204 response, no events): https://github.com/adobe/asset-compute-events-client/blob/master/lib/events.js#L398-L407

Reply:
thanks! ok 👍

Reply:
@tmathern 
> Better to choose a proper error type/message here, as I don't think we care about the original error. The whole point is to handle it and poll until ready. RIght?

thanks!, No, not testing the polling part here, but will refactor for readability and consistency.